annotation-utils/src/main/java/edu/isi/vista/annotationutils/RestoreJson.kt

```diff
@@ -1,109 +1,149 @@
-        "The path $this does not start with the prefix $prefix"
-    }
-    return this.subpath(prefix.nameCount, this.nameCount)
+package edu.isi.vista.annotationutils
+
+import edu.isi.nlp.io.OffsetIndexedCorpus
+import edu.isi.nlp.symbols.Symbol
+import edu.isi.nlp.io.DocIDToFileMappings
+import com.google.common.base.Optional
+import com.fasterxml.jackson.databind.ObjectMapper
+import edu.isi.nlp.io.OriginalTextSource
+import edu.isi.nlp.parameters.Parameters
+import java.io.File
+import java.nio.file.Files
+import java.nio.file.Path
+import java.nio.file.Paths
+import java.io.IOException;
+
+/**
+ * Puts the document text back into the 'sofaString' field inside json files created by `ExportAnnotations.kt`
+ * They are redacted from the json export because our LDC license does not permit us to distribute the full document text.
+ *
+ *
+ * The program takes one argument, a parameter file in the format described at
+ * https://github.com/isi-vista/nlp-util/blob/master/common-core-open/src/main/java/edu/isi/nlp/parameters/serifstyle/SerifStyleParameterFileLoader.java
+ *
+ * An example is provided in curated-training-annotator/sample_params/restore-text.params
+ * The required parameter values are:
+ * <ul>
+ *     <li> {@code indexDirectory} is the location of the files produced by {@code IndexFlatGigaword.java}
+(https://github.com/isi-vista/nlp-util/blob/master/nlp-core-open/src/main/java/edu/isi/nlp/corpora/gigaword/IndexFlatGigaword.java) </li>
+ *     <li> {@code gigawordDataDirectory} is the location of the gigaword text files </li>
+ *     <li> {@code aceEngDataDirectory} is the data/English directory of the ace corpus files</li>
+ *     <li> {@code inputJsonDirectory} is the location of the stripped json files produced by {@code ExportAnnotations.kt} </li>
+ *     <li> {@code restoredJsonDirectory} is where the new json files will go </li>
+ *  </ul>
+ */
+
+class RestoreJson {
+    companion object {
+        fun main(argv: Array<String>) {
+            if (argv.size != 1) {
+                throw RuntimeException("Expected a single argument, a parameter file")
+            }
+            val params = Parameters.loadSerifStyle(File(argv[0]))
+            restore(params)
+        }
+        fun restore(params: Parameters) {
+            val inputJsonDirectory = params.getExistingDirectory("inputJsonDirectory")!!
+            val outputDirectory = params.getCreatableDirectory("restoredJsonDirectory")!!
+
+            // Make objects for reading, parsing, and writing json:
+            val objectMapper = ObjectMapper()
+            val prettyPrinter = objectMapper.writerWithDefaultPrettyPrinter()
+            val gigaWordTextSource = makeTextSource(params, "gigaword")
+            val aceTextSource = makeTextSource(params, "ace")
+
+            inputJsonDirectory.walk().filter { it.isFile }.forEach { jsonFile ->
+                val filename = jsonFile.name
+                //returns docID if ace file, else returns null

```

Comment:
The following code block was changed to determine whether the document being processed from the stream is an ACE or gigaword document. The actual determination happens mostly in the `getAceDocID` function